source/simulator/core.cc

```diff
@@ -849,17 +851,46 @@ namespace aspect
         {
           Assert (is_element (*p, geometry_model->get_used_boundary_indicators()),
                   ExcInternalError());
-          VectorTools::interpolate_boundary_values (dof_handler,
-                                                    *p,
-                                                    VectorFunctionFromScalarFunctionObject<dim>(std_cxx11::bind (&BoundaryTemperature::Interface<dim>::temperature,
-                                                        std_cxx11::cref(*boundary_temperature),
-                                                        std_cxx11::cref(*geometry_model),
+
+          if (std::isnan(boundary_temperature->boundary_temperature(*p,geometry_model->representative_point(0.0))))
+            {
+              if (timestep_number == 0)
+                pcout << "WARNING: Your selected boundary temperature plugin uses the deprecated "
+                      << "interface function 'temperature(geometry_model,boundary_indicator,position)'. "
+                      << "This function will be removed in a future ASPECT "
+                      << "release. Please update your plugin NOW to the new interface "
+                      << "'boundary_temperature(boundary_indicator,position) "
+                      << "instead, and derive it from SimulatorAccess, if you need access to the "
+                      << "geometry model." << std::endl;
```

Comment:
I see. So let me rephrase your approach: you're looking for an *exceptional return path* that returns a different kind of value. My choice of words suggests already that maybe throwing (and catching) an *exception* may be the right approach ;-) This avoids the magic value and the `if-else` in the simulator (at the cost of a `try-catch`).

But why not this:
* In the base class, implement the old function and let it call the new function interface. You have all necessary parameters.
* In the base class, implement the new function and let it error out with an `Assert(false)`.
* If a class implements the old interface, this overrides the implementation in the base class and it will work (though no warning message).
* If a class implements the new interface, everything works just fine.
* In the `Simulator`, no change is necessary for now.
The only difference is that you don't get the warning. I think that's ok -- people should read the changelog.

In the end, either approach is fine with me.